internal/cluster/validator.go

```diff
@@ -458,13 +437,8 @@ func (v *clusterValidator) noCidrsOverlapping(c *clusterPreprocessContext) (Vali
 		}
 		return ValidationPending, "At least one of the CIDRs (Machine Network, Cluster Network, Service Network) is undefined."
 	}
-	if len(c.cluster.ClusterNetworks) != len(c.cluster.ServiceNetworks) {
-		// TODO MGMT-7587: Support any number of subnets
-		// Assumes that the number of cluster networks equal to the number of service networks
-		return ValidationError, "A mismatch between the number of Cluster and Service networks"
-	}
-	if err := checkCidrsOverlapping(c.cluster); err != nil {
-		return ValidationFailure, fmt.Sprintf("CIDRS Overlapping: %s.", err.Error())
+	if err := network.VerifyNoNetworkCidrOverlaps(c.cluster.ClusterNetworks, c.cluster.MachineNetworks, c.cluster.ServiceNetworks); err != nil {
+		return ValidationFailure, err.Error()
```

Comment:
The validation error messages need to be agreed upon.  They are presented by UI.  Here it is not clear what is expected as text of this error message

Reply:
Ack. I did change them, you can see the change in the tests here: https://github.com/openshift/assisted-service/pull/6071/commits/51e76b10d543c5868dcb250061c8b21250e45015#diff-854f83029709261bb4e532a5a6839b51ddeb2beb2f6e921056016a74340d06a3